### DISCUSSION TITLE
Financial value reports

### DISCUSSION THREAD

**Discussion by titanic-jb** [inventree/InvenTree#5468] [Q&A] [answered]
I have to install an inventory software for my employer.  Laws require us, to create reports on a regular basis, which calculate the value of each item present in our stock.

The main problem is, that you have to calculate the value on a first in, first out basis with their respective costs.  

Here a simple example: you order 10 pieces of item A, each costing 10 $ and put it on stock.  When only 2 pieces remain on stock, you'll order the next bunch of 10 pieces, but this time, you have to pay 20 $ for each of them.  At end of year, you have 12 pieces on stock, so the report must calculate 10  pieces at 20 $ and 2 pieces at 10 $ resulting in 12 pieces worth 240 $ in total.

Is InvenTree able to produce such reports?

I've read #5414.  From what I understand, InvenTree could be enabled, by upgrades, to generate such reports, but out of the box, it is not yet able?

**Comment by SchrodingersGat** [comment, marked as answer]
Check out the [documentation for stocktake reports](https://docs.inventree.org/en/latest/part/stocktake/). If true "cost of purchase" data is available for each stock item (as you state above) then the stocktake report will generate a total cost of purchase for stock on hand.

When a purchase order is made, the purchase cost is automatically added to the stock item when it is received into stock. This data is available for the lifecycle of each stock item.
